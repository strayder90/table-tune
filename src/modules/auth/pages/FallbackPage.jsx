import React from 'react';

const FallbackPage = () => {
    return (
        <div>
            <h1>404 - Page Not Found</h1>
            <p>The page you&#39;re looking for does not exist.</p>
        </div>
    );
};

export default FallbackPage;
