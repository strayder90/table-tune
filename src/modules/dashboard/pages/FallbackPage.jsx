import React from 'react';
import {Link} from 'react-router-dom';
import {Container, Header, Button, Image, Grid, Icon} from 'semantic-ui-react';

const FallbackPage = () => {
    return (
        <Container className='fallback-page-container'>
            <Grid centered className='fallback-page-grid'>
                <Grid.Row columns={2}>
                    <Grid.Column className='fallback-page-image'>
                        <Image src='/404.jpg' fluid size='large'/>
                    </Grid.Column>
                    <Grid.Column className='fallback-page-text'>
                        <Header as='h1' className='fallback-page-heading'>
                            A W W W... Don&#39;t worry!
                        </Header>
                        <h3>It&#39;s just a 404 Error!</h3>
                        <h3>The page you are looking for might have been removed, deleted or non existing.</h3>
                    </Grid.Column>
                    <Link to='/table-tune'>
                        <Button size='large' color='yellow' className='fallback-page-button'>
                            <Icon name='angle double left'/>
                            Go back to Home
                        </Button>
                    </Link>
                </Grid.Row>
            </Grid>
        </Container>
    );
};

export default FallbackPage;
