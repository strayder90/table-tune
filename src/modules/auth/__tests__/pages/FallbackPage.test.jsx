import React from 'react';
import '@testing-library/jest-dom';
import {render, screen} from '@testing-library/react';

import FallbackPage from '@modules/dashboard/pages/FallbackPage.jsx';

test('Displays fallback page if no match is found', () => {
    const {container} = render(<FallbackPage/>);
    expect(container).toMatchSnapshot();

    const header = screen.getByRole('heading', {name: /404 - Page Not Found/i});
    const message = screen.getByText(/The page you're looking for does not exist./i);

    expect(message).toBeInTheDocument();
    expect(header).toBeInTheDocument();
});