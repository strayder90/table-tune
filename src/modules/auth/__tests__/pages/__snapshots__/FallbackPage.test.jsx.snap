// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Displays fallback page if no match is found 1`] = `
<div>
  <div>
    <h1>
      404 - Page Not Found
    </h1>
    <p>
      The page you're looking for does not exist.
    </p>
  </div>
</div>
`;
